### PR TITLE
Implement GetModuleList over grpc

### PR DESCRIPTION
**Pull request by dimitry-**
Implement grpc GetModuleList service call

### CODE REVIEW

**Comment by dimitry-** [comment]
interesting - it builds now :)

File: OrbitGl/ProcessManager.h [hunk 1]
```diff
@@ -39,8 +41,11 @@ class ProcessManager {
   virtual void SetProcessListUpdateListener(
       const std::function<void(ProcessManager*)>& listener) = 0;
 
+  virtual outcome::result<std::vector<ModuleInfo>, std::string> GetModuleList(
+      uint32_t pid) = 0;
+
   // Get a copy of process list.
-  virtual std::vector<ProcessInfo> GetProcessList() const = 0;
+  virtual std::vector<ProcessInfo> process_list() const = 0;
```
Comment (antonrohr) [inline OrbitGl/ProcessManager.h:48]: Why this change?

Comment (dimitry-) [reply OrbitGl/ProcessManager.h:48]: I wanted to distinguish between calls that result in client-server communication and calls that just get stored data from object itself. this is probably not ideal because it is kind of implicit. I will probably change it..

Comment (antonrohr) [reply OrbitGl/ProcessManager.h:48]: okay

File: protos/module.proto [hunk 2]
```diff
@@ -0,0 +1,10 @@
+syntax = "proto3";
+
+message ModuleInfo {
+  string name = 1;
+  string path = 2;
+  uint64 address_start = 3;
+  uint64 address_end = 4;
+  uint64 size = 5;
```
Comment (dpallotti) [inline protos/module.proto:8]: What is the difference between `size` and `address_end - address_start (+1)`?

File: OrbitGl/ProcessManager.cpp [hunk 3]
```diff
@@ -55,7 +58,26 @@ void ProcessManagerImpl::SetProcessListUpdateListener(
...
+  GetModuleListRequest request;
+  GetModuleListResponse response;
+  request.set_process_id(pid);
+
+  std::unique_ptr<grpc::ClientContext> context = CreateContext();
+  grpc::Status status =
+      process_service_->GetModuleList(context.get(), request, &response);
+
+  if (!status.ok()) {
+    return status.error_message();
```
Comment (dpallotti) [inline OrbitGl/ProcessManager.cpp:74]: I assume that the outcome library understands this is a failure based on the type, but I think it would be clearer to wrap this with `outcome::failure` nonetheless.

File: OrbitGl/App.cpp [hunk 4]
```diff
@@ -671,7 +691,7 @@ outcome::result<void, std::string> OrbitApp::OnLoadCapture(
   ar.time_graph_ = GCurrentTimeGraph;
   OUTCOME_TRY(ar.Load(file_name));
 
-  m_ModulesDataView->SetProcess(Capture::GTargetProcess);
+  //  m_ModulesDataView->SetProcess(Capture::GTargetProcess);
```
Comment (dpallotti) [inline OrbitGl/App.cpp:695]: Why is this kept as a comment?

File: OrbitGl/DataManager.h [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+#include <thread>
+
+#include "ProcessData.h"
+#include "absl/container/flat_hash_map.h"
+
+// This class is responsible for storing and
+// navigating data on the client side. Note that
+// every method of this class should be called
+// on the main thread.
+class DataManager final {
```
Comment (dpallotti) [inline OrbitGl/DataManager.h:17]: So this class would contain all data displayed by the ui?

File: OrbitGl/DataManager.cpp [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+#include "DataManager.h"
+
+#include "OrbitBase/Logging.h"
+
+void DataManager::UpdateProcessInfos(
+    const std::vector<ProcessInfo>& process_infos) {
+  CHECK(std::this_thread::get_id() == main_thread_id_);
+
+  // Note that at this point the data manager does not remove old processes
+  // to do it correctly we may need to implement some callback logic here
```
Comment (dpallotti) [inline OrbitGl/DataManager.cpp:14]:
```suggestion
  // Note that at this point the data manager does not remove old processes.
  // To do it correctly we may need to implement some callback logic here
```

File: OrbitGl/App.cpp (hunk 4 above)
Comment (dimitry-) [reply OrbitGl/App.cpp:695]: I was wondering if it breaks something - I have not checked actually thanks for noticing, I will check save and load.

File: OrbitGl/DataManager.h (hunk 5 above)
Comment (dimitry-) [reply OrbitGl/DataManager.h:17]: Yes, this is the idea.

File: protos/module.proto (hunk 2 above)
Comment (dimitry-) [reply protos/module.proto:8]: The size is the file_size, it is (in most cases) different from address_start - end because of mapping gaps and .bss segment mapping rules. Maybe I should make it clear that this is file size.

Comment (dpallotti) [reply protos/module.proto:8]: Yeah, then I would rename it to make it clearer.

File: OrbitGl/ModulesDataView.cpp [hunk 7]
```diff
@@ -38,23 +38,23 @@ const std::vector<DataView::Column>& ModulesDataView::GetColumns() {
...
     case COLUMN_PATH:
-      return module->m_FullName;
+      return module->path();
     case COLUMN_ADDRESS_RANGE:
-      return module->m_AddressRange;
+      return module->address_range();
     case COLUMN_HAS_PDB:
-      return module->IsLoadable() ? "*" : "";
+      return "*";
     case COLUMN_PDB_SIZE:
```
Comment (antonrohr) [inline OrbitGl/ModulesDataView.cpp:54]: While you are at it, should those columns maybe be renamed? Here and in the UI. The naming Pdb might be confusing in a Linux context. Also, maybe remove the columns `COLUMN_HAS_PDB` completely

Comment (antonrohr) [reply OrbitGl/ModulesDataView.cpp:54]: This could also be a different PR

File: OrbitGl/ModulesDataView.cpp [hunk 8]
```diff
@@ -163,22 +158,20 @@ void ModulesDataView::OnContextMenu(const std::string& action, int menu_index,
...
   for (size_t i = 0; i < modules_.size(); ++i) {
-    std::shared_ptr<Module>& module = modules_[i];
-    std::string name = ToLower(module->GetPrettyName());
+    const ModuleData* module = modules_[i];
+    std::string module_string = absl::StrFormat(
+        "%s %s", module->address_range(), ToLower(module->path()));
 
     bool match = true;
 
     for (std::string& filterToken : tokens) {
```
Comment (antonrohr) [inline OrbitGl/ModulesDataView.cpp:173]: nit: while you are at it: 
```suggestion
    for (std::string& filter_token : tokens) {
```

File: OrbitGl/ProcessManager.h [hunk 9]
```diff
@@ -39,8 +41,11 @@ class ProcessManager {
   virtual void SetProcessListUpdateListener(
       const std::function<void(ProcessManager*)>& listener) = 0;
 
+  virtual outcome::result<std::vector<ModuleInfo>, std::string> GetModuleList(
```
Comment (antonrohr) [inline OrbitGl/ProcessManager.h:44]: Maybe the naming of this function and `process_list()` can be improved to signal which ones are blocking (aka should not be run on the main thread). Maybe something like `GetModuleListFromRemote` or maybe just a comment.

File: OrbitGl/App.cpp [hunk 10]
```diff
@@ -271,9 +269,32 @@ void OrbitApp::PostInit() {
...
+              continue;
+            }
+
+            std::shared_ptr<Process> process = std::make_shared<Process>();
+            process->SetID(info.pid());
+            process->SetName(info.name());
+            process->SetIsRemote(true);
+            // The other fields do not appear to be used at the moment.
+
+            process_map_.insert_or_assign(process->GetID(), process);
```
Comment (antonrohr) [inline OrbitGl/App.cpp:294]: suggestion: `process_map_` is only used in `FindProcessByPid(pid)` which is only used in `OnProcessSelected(pid)`, which you also touch. Maybe move this whole part into `OnProcessSelected`, then `process_map_` and `FindProcessById()` are not needed anymore and could be deleted.

File: protos/module.proto (hunk 2 above)
Comment (dimitry-) [reply protos/module.proto]: done.

File: OrbitGl/ProcessManager.h (hunk 9 above)
Comment (dimitry-) [reply OrbitGl/ProcessManager.h]: I changed the naming convention. ptal.

File: OrbitGl/ModulesDataView.cpp (hunk 7 above)
Comment (antonrohr) [reply OrbitGl/ModulesDataView.cpp:54]: okay

File: OrbitGl/ProcessManager.h (hunk 9 above)
Comment (antonrohr) [reply OrbitGl/ProcessManager.h]: I like that with using Load and Get, will change my PR accordingly

**Comment by antonrohr** [comment]
I incorporated the requested changes and fixed session loading, ptal again.
